### ISSUE TITLE
Finish transition to hex_core

### ISSUE THREAD

**Issue by wojtekmach** [hexpm/hex#738]
We're already using hex_core in a few places, this is what is left:

- [ ] Replace [Hex.API.*](https://github.com/hexpm/hex/tree/v0.20.1/lib/hex/api) with `:hex_api_*`.
- [ ] Add [progress tracking](https://github.com/hexpm/hex/blob/v0.20.1/lib/hex/api.ex#L27) to hex_core
- [ ] Convert [`Hex.HTTP`](https://github.com/hexpm/hex/blob/v0.20.1/lib/hex/http.ex) to a `:hex_http` adapter
- [ ] Update [`Hex.Repo`](https://github.com/hexpm/hex/blob/v0.20.1/lib/hex/repo.ex) to `:hex_repo` to fetch data
- [ ] Move `Hex.Crypto*` to hex_core

**Comment by cgerling** [comment]
@wojtekmach are any of the items still required to be done?

**Comment by ericmj** [comment]
@cgerling They are all up for grabs 👍

**Comment by cgerling** [comment]
Cool, I intend to create a PR to address the 1st, 3rd, and 4th items.

**Comment by cgerling** [comment]
I noticed that `:hex_repo` doesn't have some functions that `Hex.Repo` needs, like `get_docs/3`, `get_public_key/1`, and `get_installs/0`, should I add those in `hex_core` before updating `Hex.Repo`?

**Comment by ericmj** [comment]
> I noticed that :hex_repo doesn't have some functions that Hex.Repo needs, like get_docs/3, get_public_key/1, and get_installs/0, should I add those in hex_core before updating Hex.Repo?

Yes. Can you open a PR to hex_core that adds the necessary functions?

**Comment by ericmj** [comment]
Hex.Crypto is gone with OAuth, this should be done now.
